pandas/plotting/_matplotlib/tools.py

```diff
@@ -325,7 +357,15 @@ def _handle_shared_axes(axarr, nplots, naxes, nrows, ncols, sharex, sharey):
                 # have a subplot there, we can skip the layout test
                 if ax.is_first_col():
                     continue
-                if sharey or len(ax.get_shared_y_axes().get_siblings(ax)) > 1:
+                if (
+                    sharey
+                    or len(
+                        _ignore_colocated_axes(
+                            ax, ax.get_shared_y_axes().get_siblings(ax)
+                        )
+                    )
+                    > 1
```

Comment:
this expression and the one above are pretty nasty.  Would it be costly to pre-compute the _ignore_colocated_axes length to make this clearer?

Reply:
I've reworked this and it looks much better, thanks.